### DISCUSSION TITLE
Entities w/ custom models (e.g. for vehicles: a car, a train)

### DISCUSSION THREAD

**Discussion by cristian-spiescu** [Minestom/Minestom#1649] [Help] [answered]
I would like to add vehicles to my world. Such as cars or trains/subways, inspired from the [MTR mod](https://github.com/jonafanho/Minecraft-Transit-Railway) which does this pretty good.

A first step would be to "draw" some models (e.g. w/ Blockbench), pack them in a resource pack. Based [on this video](https://www.youtube.com/watch?v=aaJ8XgMAOno), I would try to use "custom model data"; e.g. 

```
pack.mcmeta Text:

{
  "pack": {
    "pack_format": 4,
    "description": "Example Resource Pack Description"
  }
}

Item Custom Model Data YOURCUSTOMITEM.json text:

{
 "parent": "item/handheld",
 "textures": {
  "layer0": "item/YOURBASEITEM"
 },

 "overrides": [
  { "predicate": {"custom_model_data": 123456}, "model": "item/YOURITEM'SJSONFILENAMEr"}

 ]
}
```

Now the **question**: after doing the above, is it enough in the Java server code, to specify somehow for my entity custom_model_data = 123456? Or should I employ a more complicated construct, e.g. using an armor stand, like [here](https://www.youtube.com/watch?v=HLFFa5mfRVI)?

Thanks in advance!

**Comment by Distractic** [comment]
In this [thread discord](https://discord.com/channels/706185253441634317/1057887362065121290), we already talked about that. A new PR will soon be made

**Comment by cristian-spiescu** [comment]
Thank you very much! Since that is an ongoing discussion, I'll continue there w/ my questions!

**Comment by iam4722202468** [comment, marked as answer]
If your goal is to add large vehicles with animations, you might want to look at this library https://github.com/SeedNexus/WorldSeedEntityEngine

It lets you use blockbench models in vanilla minecraft, and generates the resourcepack for you

**Comment by cristian-spiescu** [comment]
Thanks.
Wow! 😄 
I'll give it a try ASAP!

**Comment by cristian-spiescu** [comment]
I just ran the demo! Awesome work. Congrats! 
Now I need to browse a bit the code source, and then try to use the lib for my use case(s).
